src/array_value.cpp

```diff
@@ -257,6 +257,25 @@ ValuePtr ArrayValue::first(Env *env, ValuePtr n) {
     return array;
 }
 
+
+ValuePtr ArrayValue::drop(Env *env, ValuePtr n) {
```

Comment:
A lot of duplicated logic but let first implement everything before we refactor and optimize